gap/attr.gi

```diff
@@ -196,6 +196,96 @@ function(I, subtracted_set, size_bound)
 end
 );
 
+InstallMethod(DigraphAbsorptionProbabilities,
+"for a digraph",
+[IsDigraph],
+function(D)
+  local scc, is_sink_comp, transient_vertices, i, comp, v, sink_comps,
+        nr_transients, transient_mat, absorption_mat, neighbours, chance, w,
+        w_comp, sink_comp_index, j, N, chances_of_absorption, output, comp_no,
+        c;
+  # No vertices: trivial matrix
+  if DigraphNrVertices(D) = 0 then
+    return [];
+  fi;
+
+  scc := DigraphStronglyConnectedComponents(D);
+
+  # One SCC only: all vertices stay in it with probability 1.
+  if Length(scc.comps) = 1 then
+    return ListWithIdenticalEntries(DigraphNrVertices(D), [1]);
+  fi;
+
+  # Find the "sink components" (components from which there is no escape)
+  is_sink_comp := ListWithIdenticalEntries(Length(scc.comps), true);
+  transient_vertices := [];
+  for i in [1 .. Length(scc.comps)] do
+    comp := scc.comps[i];
+    for v in comp do
+      if ForAny(OutNeighboursOfVertex(D, v), w -> not w in comp) then
+        is_sink_comp[i] := false;
+        Append(transient_vertices, comp);
+        break;
+      fi;
+    od;
+  od;
+  sink_comps := Positions(is_sink_comp, true);
+  nr_transients := Length(transient_vertices);
```

Comment:
You could shorten this section of code by just taking the quotient of the original digraph by the strongly connected components, and then running `DigraphSinks`, and using the labels of the quotient graph to recover the actual strongly connected components.

Reply:
This is done in c28a75b3.